src/network/include/IAbstractNetworkManager.h

```diff
@@ -0,0 +1,73 @@
+#ifndef HIVE_CONNECT_IABSTRACTNETWORKMANAGER_H
+#define HIVE_CONNECT_IABSTRACTNETWORKMANAGER_H
+
+#include "cpp-common/IHashMap.h"
+#include "logger/ILogger.h"
+#include <optional>
+#include <string>
+
+enum class NetworkStatus { NotConnected = 0, Connecting, Connected };
+
+class IAbstractNetworkManager {
```

Comment:
It's not really an interface since it has some defined function. Interface are when all functions are virtuals. You could create an interface over it.

~~Though why the abstract, why not just have the interface and then implementation directly on the NetworkManager as usual?~~ I see it's to avoid code duplication for the esp32 port right?

Reply:
Yep, since the functions to register are identical, it avoids code duplication and can be tested on both platforms